Source/WebCore/rendering/TextBoxPainter.cpp

```diff
@@ -368,6 +368,20 @@ void TextBoxPainter<TextBoxPath>::paintForegroundAndDecorations()
     } else {
         // Coalesce styles of adjacent marked texts to minimize the number of drawing commands.
         auto coalescedStyledMarkedTexts = StyledMarkedText::coalesceAdjacentWithEqualForeground(styledMarkedTexts);
+        bool isHighlights = false;
+
+        for (auto& element : coalescedStyledMarkedTexts) {
+            if (!isHighlights && element.type == 6) {
```

Comment:
The `!isHighlights` check isn't necessary, right? You're breaking out as soon as you set it anyway.

```suggestion
            if (element.type == MarkedText::Type::Highlight) {
```

Reply:
Also, we should consider making `MarkedText::Type` an `enum class` if you have time! (not in this PR, of course)